src/complex/Parameters/GeometrySelectionParameter.cpp

```diff
@@ -111,7 +113,9 @@ Result<> GeometrySelectionParameter::validatePath(const DataStructure& dataStruc
     return {};
   }
 
-  return MakeErrorResult(-4, fmt::format("Object at path '{}' is not an accepted geometry type.", value.toString()));
+  return complex::MakeErrorResult(complex::FilterParameter::Constants::k_Validate_AllowedType_Error,
+                                  fmt::format("{}Geometry at path '{}' was of type '{}', but only {} are allowed", prefix, value.toString(), abstractGeometry->getGeometryTypeAsString(),
+                                              fmt::join(AbstractGeometry::StringListFromGeometryType(m_AllowedTypes), ",")));
```

Comment:
If you add `#include <fmt/ranges.h>`, you can just pass the `std::set<std::string>` directly.

Reply:
Addressed.